packages/dashboard-hardhat-plugin/test/fixture-projects/hardhat-project-incompatible/.gitignore

```diff
@@ -0,0 +1 @@
+/typechain-types
```

Comment:
Hey @cds-amal, if you have moment, do you mind trying this again against the latest on this branch? It's not modifying on my side now and I "think" it might have been due to a mismatch between the test's contract (`SimpleStorage.sol`) and the accompanying artifacts. 

Also keeping these files is necessary (at least afaik) given it's testing for an invalid build-info entry in `artifacts/build-info/b4ddd0c0e9f2d6be7fc46234ad4015a9.json` and there's no way to generate it via a compilation.

Reply:
@kevinbluer, confirmed the tests mutate source controlled assets. See session:

```console
╭─amal@faifai  ~/work/truffle/packages/dashboard-hardhat-plugin ‹node-18.12.1› (feat/dashboard-hardhat-plugin)
╰─$ git st
On branch feat/dashboard-hardhat-plugin
Your branch is up to date with 'origin/feat/dashboard-hardhat-plugin'.

nothing to commit, working tree clean
╭─amal@faifai  ~/work/truffle/packages/dashboard-hardhat-plugin ‹node-18.12.1› (feat/dashboard-hardhat-plugin)
╰─$ yarn test
--- snip ---
  Truffle dashboard hardhat plugin compilation tests
    Compiling with an incompatible Hardhat version
      ✔ should fail if Hardhat version is below 2.10.1 (354ms)
    Compiling with incompatible hardhat build info
Error HH12: Trying to use a non-local installation of Hardhat, which is not supported.
Please install Hardhat locally using npm or Yarn, and try again.
For more info go to https://hardhat.org/HH12 or run Hardhat with --show-stack-traces
      ✔ should fail when incompatible hardhat build info (e.g. hh-sol-build-info-2) is detected (854ms)
    Compiling when no valid Hardhat project
      ✔ should fail when attempting to compile


  3 passing (1s)

✨  Done in 3.42s.
╭─amal@faifai  ~/work/truffle/packages/dashboard-hardhat-plugin ‹node-18.12.1› (feat/dashboard-hardhat-plugin*)
╰─$ git st
On branch feat/dashboard-hardhat-plugin
Your branch is up to date with 'origin/feat/dashboard-hardhat-plugin'.

Changes not staged for commit:
	modified:   test/fixture-projects/hardhat-project-incompatible/artifacts/build-info/b4ddd0c0e9f2d6be7fc46234ad4015a9.json
	modified:   test/fixture-projects/hardhat-project-incompatible/cache/solidity-files-cache.json

no changes added to commit
```

Reply:
Thanks for confirming @cds-amal! It should hopefully be negated by this https://github.com/trufflesuite/truffle/pull/5927#discussion_r1128666489 (which also now ignores them too).

Reply:
Resolving given it's now addressed with the refactor in https://github.com/trufflesuite/truffle/pull/5927/commits/58a2f0f38d34dda8e2c37689083994615f2563f5#diff-14b74f2ba79d9e8e2f2cee6320711cc35827ca6d2bf09b1ad1b87e10ed4f052b 👍